app/models/spree/order.rb

```diff
@@ -734,16 +734,5 @@ def update_adjustment!(adjustment)
       adjustment.update_adjustment!(force: true)
       updater.update_totals_and_states
     end
-
-    # object_params sets the payment amount to the order total, but it does this
-    # before the shipping method is set. This results in the customer not being
-    # charged for their order's shipping. To fix this, we refresh the payment
-    # amount here.
-    def set_payment_amount!
-      update_totals
-      return unless pending_payments.any?
-
-      pending_payments.first.update_attribute :amount, total
-    end
```

Comment:
This was a bit funky to begin with, but it's not the `Order` class's responsibility any more, which I think is an improvement.

The purpose of this `set_payment_amount!` method here could more accurately be described as: "retroactively _reset_ the order's payment amount part-way through checkout processing".